includes/api/class-wc-rest-product-reviews-controller.php

```diff
@@ -0,0 +1,271 @@
+		$data = array(
+			'id'             => (int) $review->comment_ID,
+			'date_created'   => wc_rest_prepare_date_response( $review->comment_date_gmt ),
+			'review'         => $review->comment_content,
+			'rating'         => get_comment_meta( $review->comment_ID, 'rating', true ),
+			'reviewer_name'  => $review->comment_author,
+			'reviewer_email' => $review->comment_author_email,
+			'verified'       => wc_review_is_from_verified_owner( $review->comment_ID ),
+		);
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $review, $request ) );
+
+		/**
+		 * Filter webhook delivery object returned from the REST API.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param WP_Comment       $review   Product review object used to create response.
+		 * @param WP_REST_Request  $request  Request object.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_product_review', $response, $review, $request );
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param WP_Comment $review Product review object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return array Links for the given product review.
+	 */
+	protected function prepare_links( $review, $request ) {
+		$product_id = (int) $request['product_id'];
+		$base       = str_replace( '(?P<product_id>[\d]+)', $product_id, $this->rest_base );
+		$links      = array(
+			'self' => array(
+				'href' => rest_url( sprintf( '/%s/%s/%d', $this->namespace, $base, $review->comment_ID ) ),
+			),
+			'collection' => array(
+				'href' => rest_url( sprintf( '/%s/%s', $this->namespace, $base ) ),
+			),
+			'up' => array(
+				'href' => rest_url( sprintf( '/%s/products/%d', $this->namespace, $product_id ) ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get the Product Review's schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'product_review',
+			'type'       => 'object',
+			'properties' => array(
+				'id' => array(
+					'description' => __( 'Unique identifier for the resource.', 'woocommerce' ),
+					'type'        => 'integer',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'date_created' => array(
+					'description' => __( "The date the review was created, in the site's timezone.", 'woocommerce' ),
+					'type'        => 'date-time',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'rating' => array(
+					'description' => __( 'Review rating (0 to 5).', 'woocommerce' ),
+					'type'        => 'string',
```

Comment:
Should this be integer instead of string? For ratings 0-5. Or float (I can't remember if partial rates are allowed).